sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Tests/ScopeTests.cs

```diff
@@ -0,0 +1,147 @@
+                _httpRequest.Headers.Add("Www-Authenticate", headerValue);
+
+                var client = await ACRTestUtil.GetCredentialsAsync(context, ACRTestUtil.ManagedTestRegistryForChanges);
+                string actualScope = client.GetScopeFromHeaders(_httpRequest.Headers);
+                Assert.Equal(expectedScope, actualScope);
+            }
+        }
+
+        public static IEnumerable<object[]> TestData()
+        {
+            yield return new object[] { "Bearer realm=\"https://myregistry.azurecr-test.io/oauth2/token\",service=\"myregistry.azurecr-test.io\",scope=\"foo:bar:baz,abc\",Basic xyz=123", "foo:bar:baz,abc"};
+            yield return new object[] { "Bearer realm=\"https://myregistry.azurecr-test.io/oauth2/token\",service=\"myregistry.azurecr-test.io\",scope=\"foo:bar:baz,abc\", Basic xyz=123", "foo:bar:baz,abc"};
+            yield return new object[] { "Bearer realm=\"https://myregistry.azurecr-test.io/oauth2/token\",service=\"myregistry.azurecr-test.io\",scope=\"foo:bar:baz,abc\"", "foo:bar:baz,abc" };
+            yield return new object[] { "Bearer realm=\"https://myregistry.azurecr-test.io/oauth2/token\",service=\"myregistry.azurecr-test.io\",scope=\"foo:bar:baz\"", "foo:bar:baz" };
+        }
+
+        private static readonly V2Manifest ExpectedV2ManifestProd = new V2Manifest()
+        {
+            SchemaVersion = 2,
+            MediaType = ACRTestUtil.MediatypeV2Manifest,
+            Config = new Descriptor
+            {
+                MediaType = ACRTestUtil.MediatypeV1Manifest,
+                Size = 5635,
+                Digest = "sha256:16463e0c481e161aabb735437d30b3c9c7391c2747cc564bb927e843b73dcb39"
+            },
+            Layers = new List<Descriptor>
+            {
+                new Descriptor
+                {
+                    MediaType = "application/vnd.docker.image.rootfs.diff.tar.gzip",
+                    Size = 2789742,
+                    Digest = "sha256:0503825856099e6adb39c8297af09547f69684b7016b7f3680ed801aa310baaa"
+                },
+                new Descriptor
+                {
+                    MediaType = "application/vnd.docker.image.rootfs.diff.tar.gzip",
+                    Size = 3174556,
+                    Digest = "sha256:7bf5420b55e6bbefb64ddb4fbb98ef094866f3a3facda638a155715ab6002d9b"
+                },
+                new Descriptor
+                {
+                    MediaType = "application/vnd.docker.image.rootfs.diff.tar.gzip",
+                    Size = 344,
+                    Digest = "sha256:1beb2aaf8cf93eacf658fa7f7f10f89ccec1838d1ac643a273345d4d0bc813a8"
+                }
+            }
+        };
+
+        private void VerifyManifest(Manifest baseManifest, Manifest actualManifest)
+        {
+            Assert.Equal(baseManifest.GetType(), actualManifest.GetType());
+            Assert.Equal(baseManifest.SchemaVersion, actualManifest.SchemaVersion);
+
+            if (baseManifest.GetType() == typeof(V2Manifest))
```

Comment:
Use pattern matching with `is`: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/is
Example: `baseManifest is V2Manifest v2Manifest`